components/brave_new_tab_ui/reducers/new_tab_reducer.ts

```diff
@@ -116,26 +116,41 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       }
       break
 
-    case types.CUSTOM_BACKGROUND_UPDATED:
+    case types.BACKGROUND_UPDATED:
       // While customizing background, background has
       // custom or brave default background. Branded wallpaper will
       // be visible after reloading or new NTP after changing custom
       // background option.
       state = {
         ...state,
-        brandedWallpaper: undefined
+        brandedWallpaper: undefined,
+        backgroundWallpaper: undefined
       }
 
-      // Empty custom bg url means using brave background.
-      const customBackground = payload.customBackground as CustomBackground
-      const url = customBackground.url.url
-      const color = customBackground.color
-      const random = customBackground.useRandomItem
-      if (url) {
-        state.backgroundWallpaper = { type: 'image', wallpaperImageUrl: url }
-      } else if (color) {
-        state.backgroundWallpaper = random ? backgroundAPI.randomColorBackground(color) : { type: 'color', wallpaperColor: color, random }
-      } else {
+      const background = payload.background as Background
+      if (background?.custom) {
+        const url = background.custom.url.url
+        const color = background.custom.color
+        const random = background.custom.useRandomItem
+        if (url) {
+          state.backgroundWallpaper = { type: 'image', wallpaperImageUrl: url }
+        } else if (color) {
+          state.backgroundWallpaper = random ? backgroundAPI.randomColorBackground(color) : { type: 'color', wallpaperColor: color, random }
+        }
+      }
+
+      if (!state.backgroundWallpaper && background?.brave) {
+        const braveBackground: NewTab.BraveBackground = {
```

Comment:
Optional:
If you merge the declaration and assignment, the type should be inferred (because of the `type: 'brave'`)

Reply:
Sorry, but do you mean that `const background = { ... } ` ? I'm not sure if it's what you mean but when I try this, `background` is infered as `any`. Is it what it supposed to be? 

<img width="402" alt="image" src="https://user-images.githubusercontent.com/5474642/187395119-11dd5861-b936-40ff-92c6-d8bd827fc563.png">


---

Anyway, the following code without type works well :)
```        
        state.backgroundWallpaper = {
          type: 'brave',
          author: background.brave.author,
          link: background.brave.link.url,
          wallpaperImageUrl: background.brave.imageUrl.url,
          random: false
        }
```

Reply:
Yeah, the second example is what I was thinking :smile: